pkg/client/messaging/client.go

```diff
@@ -95,6 +106,24 @@ func SendByDestination(destination *service.Destination) SendMessageOpions {
 	}
 }
 
+// WaitForResponse option to set message response type.
+// Message reply will wait for the response of this message type and matching thread ID.
+func WaitForResponse(responseType string) SendMessageOpions {
+	return func(opts *sendMsgOpts) {
+		opts.responseMsgType = responseType
+	}
+}
+
+// WithContext is context for await for reply operation.
```

Comment:
A fresh context should be used for each operation. I'm a bit concerned about the option pattern. Can you clarify?

Reply:
we need context/timeout only when someone uses await reply option, if someone doesn't pass it then we use context with default timeout(20 seconds)